worlds/dark_souls_3/__init__.py

```diff
@@ -0,0 +1,281 @@
+            # Specific items will be included in the item pool under certain conditions. See generate_basic
+            if name != "Basin of Vows":
+                self.world.itempool += [self.create_item(name)]
+
+    def generate_early(self):
+        pass
+
+    def set_rules(self) -> None:
+
+        # Define the access rules to the entrances
+        set_rule(
+            self.world.get_entrance("Goto Bell Tower", self.player),
+            lambda state: state.has("Mortician's Ashes", self.player))
+        set_rule(
+            self.world.get_entrance("Goto Undead Settlement", self.player),
+            lambda state: state.has("Small Lothric Banner", self.player))
+        set_rule(
+            self.world.get_entrance("Goto Lothric Castle", self.player),
+            lambda state: state.has("Basin of Vows", self.player))
+        set_rule(
+            self.world.get_location("HWL: Soul of the Dancer", self.player),
+            lambda state: state.has("Basin of Vows", self.player))
+        set_rule(
+            self.world.get_entrance("Goto Irithyll of the boreal", self.player),
+            lambda state: state.has("Small Doll", self.player))
+        set_rule(
+            self.world.get_entrance("Goto Archdragon peak", self.player),
+            lambda state: state.has("Path of the Dragon Gesture", self.player))
+        set_rule(
+            self.world.get_entrance("Goto Profaned capital", self.player),
+            lambda state: state.has("Storm Ruler", self.player))
+        set_rule(
+            self.world.get_entrance("Goto Grand Archives", self.player),
+            lambda state: state.has("Grand Archives Key", self.player))
+        set_rule(
+            self.world.get_entrance("Goto Kiln Of The First Flame", self.player), lambda state:
+            state.has("Cinders of a Lord - Abyss Watcher", self.player) and
+            state.has("Cinders of a Lord - Yhorm the Giant", self.player) and
+            state.has("Cinders of a Lord - Aldrich", self.player) and
+            state.has("Cinders of a Lord - Lothric Prince", self.player))
+
+        self.world.completion_condition[self.player] = lambda state: \
+            state.has("Cinders of a Lord - Abyss Watcher", self.player) and \
+            state.has("Cinders of a Lord - Yhorm the Giant", self.player) and \
+            state.has("Cinders of a Lord - Aldrich", self.player) and \
+            state.has("Cinders of a Lord - Lothric Prince", self.player)
+
+    def generate_basic(self):
+        # Depending on the specified option, add the Basin of Vows to a specific location or to the item pool
+        item = self.create_item("Basin of Vows")
+        if self.world.late_basin_of_vows[self.player]:
+            self.world.get_location("IBV: Soul of Pontiff Sulyvahn", self.player).place_locked_item(item)
+        else:
+            self.world.itempool += [item]
+
+        # Fill item pool with additional items
```

Comment:
This doesn't do anything if the item pool is already larger than the location pool which it apparently is with the default settings.

Reply:
I removed this part of the code as their is no more option to enter in this loop